fetch/api/cors/data-url-iframe.html

```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
+<html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body></body>
+<script>
+
+const createDataUrlIframe = (url, cors) => {
+  const iframe = document.createElement("iframe");
+  const fetchURL = new URL(url, location.href) +
+      `${cors === 'null-origin'
+           ? '?pipe=header(Access-Control-Allow-Origin, null)' : ''}`;
+  const tag_name = 'script';
+  iframe.src =
+      `data:text/html, <${tag_name}>` +
+      `async function test() {` +
+      `  let allowed = true;` +
+      `  try {` +
+      `    await fetch('${fetchURL}');` +
+      `  } catch (e) {` +
+      `    allowed = false;` +
+      `  }` +
+      `  parent.postMessage({allowed}, '*');` +
+      `}` +
+      `test(); </${tag_name}>`;
+  return iframe;
+};
+
+const fetch_from_data_url_iframe_test =
+    (url, cors, expectation, description) => {
+  promise_test(async () => {
+    const iframe = createDataUrlIframe(url, cors);
+    document.body.appendChild(iframe);
+    const msgEvent = await new Promise(resolve => window.onmessage = resolve);
+    assert_equals(msgEvent.data.allowed ? 'allowed' : 'rejected', expectation);
+  }, description);
+};
+
+fetch_from_data_url_iframe_test(
+  '../resources/top.txt',
+  'no-cors',
+  'rejected',
+  'fetching "top.txt" with no CORS should be rejected.'
```

Comment:
This is a tad ambiguous as the request mode is "cors" (not "no-cors").

Reply:
I changed 'no-cors' to 'same-origin' as this corresponds to 'null-origin'.

Reply:
Well, it's also not same-origin, right? A data: URL frame/worker has an opaque origin.

Reply:
Maybe just saying 'default' is better?

Reply:
Or something like acao-omitted and acao-null. Sorry, don't have great suggestions either. (Either way, it's approved, if you want to land and move on, feel free.)

Reply:
Thanks for the suggestions. I prefer 'acao-omitted'.